frontend/src/components/NumberInput/NumberInput.tsx

```diff
@@ -0,0 +1,97 @@
+import { useRef } from 'react'
+import { BiMinus, BiPlus } from 'react-icons/bi'
+import {
+  Box,
+  chakra,
+  Divider,
+  forwardRef,
+  NumberInputProps as ChakraNumberInputProps,
+  useFormControlProps,
+  useMultiStyleConfig,
+  useNumberInput,
+} from '@chakra-ui/react'
+
+import IconButton from '../IconButton'
+
+export interface NumberInputProps extends ChakraNumberInputProps {
+  /**
+   * Whether the input is in a prefilled state.
+   */
+  isPrefilled?: boolean
+  /**
+   * Whether the input is in a success state.
+   */
+  isSuccess?: boolean
+  /**
+   * Whether to show the increment and decrement steppers. Defaults to true.
+   */
+  showSteppers?: boolean
+}
+
+export const NumberInput = forwardRef<NumberInputProps, 'div'>(
+  (
+    {
+      showSteppers = true,
+      clampValueOnBlur = false,
+      isSuccess,
+      isPrefilled,
+      ...props
+    },
+    ref,
+  ) => {
+    const styles = useMultiStyleConfig('NumberInput', {
+      ...props,
+      isSuccess,
+      isPrefilled,
+    })
+
+    const stepperWrapperRef = useRef<HTMLDivElement | null>(null)
+
+    const controlProps = useFormControlProps(props)
+    const {
+      htmlProps,
+      getInputProps,
+      getIncrementButtonProps,
+      getDecrementButtonProps,
+    } = useNumberInput({ ...controlProps, clampValueOnBlur })
+
+    const inputProps = getInputProps({ placeholder: props.placeholder })
+    const incProps = getIncrementButtonProps()
+    const decProps = getDecrementButtonProps()
+
+    const inputEndPadding = showSteppers
+      ? stepperWrapperRef.current?.offsetWidth
+      : undefined
+
+    return (
+      <Box {...htmlProps} ref={ref} __css={styles.root}>
+        <chakra.input
```

Comment:
why are we using chakra's internal `input` here?

Reply:
the `Input` component performs more transformation of `inputProps`, removing some important variables such as `aria-invalid`, etc. Resulting in some styles not being provided. I'll add a comment

Reply:
Possible to patch upstream, i will open an issue if I have time....